log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -711,7 +707,8 @@ public Configuration setConfiguration(final Configuration config) {
                 prev.stop();
             }
 
-            firePropertyChangeEvent(new PropertyChangeEvent(this, PROPERTY_CONFIG, prev, config));
+            final var event = new ConfigurationChangeEvent(this, prev, config);
+            configurationChangeListeners.forEach(listener -> listener.onChange(event));
```

Comment:
Probably we should add a `try ... catch` block to prevent bugs in the listeners from breaking Log4j.

Reply:
I suppose that makes sense as an improvement.